wandb/sdk/artifacts/artifact.py

```diff
@@ -1087,14 +1088,27 @@ def __setitem__(self, name: str, item: data_types.WBValue) -> ArtifactManifestEn
 
     @contextlib.contextmanager
     def new_file(
-        self, name: str, mode: str = "w", encoding: Optional[str] = None
+        self,
+        name: str,
+        mode: str = "w",
+        encoding: Optional[str] = None,
+        skip_cache: Optional[bool] = False,
+        policy: Optional[str] = "mutable",
     ) -> Generator[IO, None, None]:
         """Open a new temporary file and add it to the artifact.
 
         Arguments:
             name: The name of the new file to add to the artifact.
             mode: The file access mode to use to open the new file.
             encoding: The encoding used to open the new file.
+            skip_cache: If set to `True`, W&B will not copy/move files to the cache while uploading
```

Comment:
This would be a good candidate for skipping the cache by default; we shouldn't actually do that because it's a behavior change, but it's the kind of thing I would add to the list of changes we would include in a breaking release.

Reply:
moved the staging changes to a separate pr instead